From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rtlsdrblog <admin@rtl-sdr.com>
Date: Sun, 27 Aug 2023 15:12:55 +1200
Subject: [PATCH] follow keenerd and don't stop the i2c repeater on tuner
 gain/freq change. Fixes slow tune/gain changes on AMD devices apparently

---
 src/librtlsdr.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/librtlsdr.c b/src/librtlsdr.c
index 24480af..4ca9692 100644
--- a/src/librtlsdr.c
+++ b/src/librtlsdr.c
@@ -906,11 +906,12 @@ int rtlsdr_set_center_freq(rtlsdr_dev_t *dev, uint32_t freq)
 	}
 
 	if (dev->direct_sampling) {
+		rtlsdr_set_i2c_repeater(dev, 0);
 		r = rtlsdr_set_if_freq(dev, freq);
 	} else if (dev->tuner && dev->tuner->set_freq) {
 		rtlsdr_set_i2c_repeater(dev, 1);
 		r = dev->tuner->set_freq(dev, freq - dev->offs_freq);
-		rtlsdr_set_i2c_repeater(dev, 0);
+		/*rtlsdr_set_i2c_repeater(dev, 0);*/
 	}
 
 	if (!r)
@@ -1059,7 +1060,7 @@ int rtlsdr_set_tuner_gain(rtlsdr_dev_t *dev, int gain)
 	if (dev->tuner->set_gain) {
 		rtlsdr_set_i2c_repeater(dev, 1);
 		r = dev->tuner->set_gain((void *)dev, gain);
-		rtlsdr_set_i2c_repeater(dev, 0);
+		/*rtlsdr_set_i2c_repeater(dev, 0);*/
 	}
 
 	if (!r)
@@ -1088,7 +1089,7 @@ int rtlsdr_set_tuner_if_gain(rtlsdr_dev_t *dev, int stage, int gain)
 	if (dev->tuner->set_if_gain) {
 		rtlsdr_set_i2c_repeater(dev, 1);
 		r = dev->tuner->set_if_gain(dev, stage, gain);
-		rtlsdr_set_i2c_repeater(dev, 0);
+		/*rtlsdr_set_i2c_repeater(dev, 0);*/
 	}
 
 	return r;
@@ -1104,7 +1105,7 @@ int rtlsdr_set_tuner_gain_mode(rtlsdr_dev_t *dev, int mode)
 	if (dev->tuner->set_gain_mode) {
 		rtlsdr_set_i2c_repeater(dev, 1);
 		r = dev->tuner->set_gain_mode((void *)dev, mode);
-		rtlsdr_set_i2c_repeater(dev, 0);
+		/*rtlsdr_set_i2c_repeater(dev, 0);*/
 	}
 
 	return r;
-- 
2.45.1

